pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/SegmentCommitter.java

```diff
@@ -0,0 +1,108 @@
+package org.apache.pinot.core.data.manager.realtime;
+
+import java.io.File;
+import org.apache.pinot.common.protocols.SegmentCompletionProtocol;
+import org.apache.pinot.core.segment.index.loader.IndexLoadingConfig;
+import org.apache.pinot.server.realtime.ServerSegmentCompletionProtocolHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Segment committer class commits realtime segments through both split commit and regular commit pathways.
+ */
+public class SegmentCommitter {
+
+  private boolean _isSplitCommit;
+  private String _segmentNameStr;
+  private ServerSegmentCompletionProtocolHandler _protocolHandler;
+  private IndexLoadingConfig _indexLoadingConfig;
+  private SegmentCompletionProtocol.Response _prevResponse;
+  private SegmentCompletionProtocol.Request.Params _params;
+
+  private final Logger SEGMENT_LOGGER;
+
+  public SegmentCommitter(boolean isSplitCommit, String segmentNameStr, ServerSegmentCompletionProtocolHandler protocolHandler,
+      IndexLoadingConfig indexLoadingConfig, SegmentCompletionProtocol.Response prevResponse, SegmentCompletionProtocol.Request.Params params) {
+    _isSplitCommit = isSplitCommit;
+    _segmentNameStr = segmentNameStr;
+    _protocolHandler = protocolHandler;
+    _indexLoadingConfig = indexLoadingConfig;
+    _prevResponse = prevResponse;
+    _params = params;
+
+    SEGMENT_LOGGER = LoggerFactory.getLogger(LLRealtimeSegmentDataManager.class.getName() + "_" + _segmentNameStr);
```

Comment:
this is already computed in LLRealtimeSegmentDataManager. Can we pass this in the constructor?